pkg/autoscaler/metrics/collector.go

```diff
@@ -406,3 +406,28 @@ func (c *collection) close() {
 	close(c.stopCh)
 	c.grp.Wait()
 }
+
+// add adds the stats from `src` to `dst`.
+func (dst *Stat) add(src Stat) {
+	dst.AverageConcurrentRequests += src.AverageConcurrentRequests
+	dst.AverageProxiedConcurrentRequests += src.AverageProxiedConcurrentRequests
+	dst.RequestCount += src.RequestCount
+	dst.ProxiedRequestCount += src.ProxiedRequestCount
+}
+
+// average reduces the aggregate stat from `sample` pods to an averaged one over
+// `total` pods.
+//
+// Assumption: A particular pod can stand for other pods, i.e. other pods
+// have similar concurrency and QPS.
+//
+// Hide the actual pods behind scraper and send only one stat for all the
+// customer pods per scraping. The pod name is set to a unique value, i.e.
+// scraperPodName so in autoscaler all stats are either from activator or
+// scraper.
+func (dst *Stat) average(sample, total float64) {
```

Comment:
I know we didn't check whether `sample` is 0 before, where `sample` equals to `readyPods` in the only usage. Is it guaranteed that `readyPods` should be 0? Should we add a non-zero check here or add a comment to indicate the input `sample` shouldn't be 0?

Reply:
Given it's an internal method, I think it's fine. I'll comment though.

Reply:
I.e if there's a need to make the method public, we can do that at that time.